### ISSUE TITLE
OneHotEncoder creates extra column when handle_missing = 'error'

### ISSUE THREAD

**Issue by alexandersmedley** [scikit-learn-contrib/category_encoders#223]
I am using the OneHotEncoder like this : 
```
# One Hot 
enc_one_hot = ce.OneHotEncoder(cols = cat_features, handle_missing = 'error')
X_train = enc_one_hot.fit_transform(X_train)
X_test = enc_one_hot.transform(X_test)
```
Using handle_missing = 'error' generates for each categorical variable an additional column 'variable_-2' the same way handle_missing = 'indicator' does. 

There is no missing value in the df I am encoding, so the 'error' option doesn't activate an error. The additional column is full of zeros for all the categorical variables. 

The documentation mentions 'indicator' will add a column but doesn't mention 'error' does. Could this be a bug in the error option ? I'm guessing the excepted output for 'error' would only be activating the error and not adding a column. 

Please let me know if you need me to provide the data on which I observed this behaviour. 

PS : Using handle_missing = 'error' with BinaryEncoder or TargetEncoder does not add any 'variable_-2' column.

**Comment by janmotl** [comment]
This appears to be an undesired behaviour.

**Comment by PaulWestenthanner** [comment]
This was also fixed by #322
